source/extensions/filters/http/cache/cache_headers_utils.h

```diff
@@ -98,6 +100,37 @@ class CacheHeadersUtils {
    */
   static absl::optional<uint64_t> readAndRemoveLeadingDigits(absl::string_view& str);
 };
+
+class VaryHeader {
+public:
+  // Checks if the HeaderEntry contains a valid value in the Vary header.
+  static bool isAllowed(const Http::ResponseHeaderMap& headers);
+
+  // Checks if the HeaderEntry contains a populated Vary header.
+  static bool hasVary(const Http::ResponseHeaderMap& headers);
+
+  // Creates a single string combining the values of the varied headers from entry_headers.
+  static std::string createVaryKey(const Http::HeaderEntry* vary_header,
+                                   const std::vector<const Http::HeaderEntry*>& entry_headers);
+
+  // Parses the header names that are in the Vary header value.
+  static std::vector<std::string> parseHeaderValue(const Http::HeaderEntry* vary_header);
+
+  // Returns a vector with the headers that can be varied from the request.
+  static std::vector<const Http::HeaderEntry*>
+  possibleVariedHeaders(const Http::RequestHeaderMap& request_headers);
+
+private:
+  // Parses the allowlist of header values that can be used to create varied responses.
+  static absl::flat_hash_set<std::string> parseAllowlist();
+
+  // Set containing the headers that can be used to vary responses.
+  inline static const absl::flat_hash_set<std::string> allowed_headers_ = parseAllowlist();
```

Comment:
Since we're only accepting `accept-encoding` for now, would a TODO comment suffice to make this explicit?

Reply:
My concern is that fixing this will change the structure of much of the code in this PR. You could have the relevant functions take the allowlist as a parameter.

Reply:
That's a good idea, I'll make the `parseAllowlist` public, and the other functions have the set as a parameter.